File: app/src/main/java/com/mytlogos/enterprise/background/room/BaseDao.java
package com.mytlogos.enterprise.background.room;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Update;

public interface BaseDao<T> {
    @Update
    void update(T t);

    @Delete
    void delete(T t);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(T t);
}

File: app/src/main/java/com/mytlogos/enterprise/background/room/BaseDao.kt
package com.mytlogos.enterprise.background.room

import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Update

interface BaseDao<T> {
    @Update
    suspend fun update(t: T)

    @Delete
    suspend fun delete(t: T)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(t: T)
}